Source/Meadow.Core/Cloud/MeadowCloudUpdateService.cs

```diff
@@ -230,31 +231,56 @@ private async Task DownloadProc(UpdateMessage message)
                 // Note: this is infrequently called, so we don't want to follow the advice of "use one instance for all calls"
                 using (var httpClient = new HttpClient())
                 {
-                    if (_connectionService.Settings.UseAuthentication)
+                    using (var request = new HttpRequestMessage(HttpMethod.Get, destination))
                     {
-                        httpClient.DefaultRequestHeaders.Authorization = _connectionService.CreateAuthenticationHeaderValue();
-                    }
-
-                    // Configure the HTTP range header to indicate resumption of partial download, starting from 
-                    // the 'totalBytesDownloaded' byte position and extending to the end of the content.
-                    httpClient.DefaultRequestHeaders.Range = new System.Net.Http.Headers.RangeHeaderValue(totalBytesDownloaded, null);
+                        if (_connectionService.Settings.UseAuthentication)
+                        {
+                            request.Headers.Authorization = _connectionService.CreateAuthenticationHeaderValue();
+                        }
 
-                    using (var stream = await httpClient.GetStreamAsync(destination))
-                    {
-                        using (var fileStream = Store.GetUpdateFileStream(message.ID))
+                        // Get the actual update file size, which might be larger than the expected
+                        // if this OTA requires an OS update
+                        var response = await httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+                        var contentLength = response.Content.Headers.ContentLength;
+                        if (contentLength.HasValue)
+                        {
+                            message.FileSize = contentLength.Value;
```

Comment:
This will be different based on the Range header (if the Range section is placed above this). It will be the number of bytes remaining, not the total file size. I'm not sure if that matters, but just a heads up.

If you care about total file size, you'll have to have a variable that exists outside of this for-loop and set it the first time it attempts to download the file (when `retryCount` is `0`).

If you only care about bytes remaining, then this is fine. But it won't be representative of the total files size, _if_ this retry loop loops.